aviary/subsystems/propulsion/hamilton_standard.py

```diff
@@ -0,0 +1,932 @@
+            outputs['thrust_coefficient'][i_node] = ct
+            outputs['comp_tip_loss_factor'][i_node] = xft
+
+
+class PostHamiltonStandard(om.ExplicitComponent):
+    """
+    Post-process after HamiltonStandard run to get thrust and compressibility
+    """
+
+    def initialize(self):
+        self.options.declare('num_nodes', types=int)
+
+    def setup(self):
+        nn = self.options['num_nodes']
+
+        self.add_input(Aircraft.Engine.PROPELLER_DIAMETER, val=0.0, units='ft')
+        self.add_input(Dynamic.Mission.INSTALLATION_LOSS_FACTOR,
+                       val=np.zeros(nn), units='unitless')
+        self.add_input('thrust_coefficient', val=np.zeros(nn), units='unitless')
+        self.add_input('comp_tip_loss_factor', val=np.zeros(nn), units='unitless')
+        self.add_input(Dynamic.Mission.PROPELLER_TIP_SPEED,
+                       val=np.zeros(nn), units='ft/s')
+        self.add_input('density_ratio', val=np.zeros(nn), units='unitless')
+        self.add_input('adv_ratio', val=np.zeros(nn), units='unitless')
+        self.add_input('power_coefficient', val=np.zeros(nn), units='unitless')
+
+        self.add_output('thrust_coefficient_comp_loss',
+                        val=np.zeros(nn), units='unitless')
+        self.add_output('Thrust', val=np.zeros(nn), units='lbf')
+        # keep them for reporting but don't seem to be required
+        self.add_output('propeller_efficiency', val=np.zeros(nn), units='unitless')
+        self.add_output('install_efficiency', val=np.zeros(nn), units='unitless')
+
+    def setup_partials(self):
+        arange = np.arange(self.options['num_nodes'])
+
+        self.declare_partials('thrust_coefficient_comp_loss', [
+            'thrust_coefficient',
+            'comp_tip_loss_factor',
+        ], rows=arange, cols=arange)
+        self.declare_partials('Thrust', [
+            'thrust_coefficient',
+            'comp_tip_loss_factor',
+            Dynamic.Mission.PROPELLER_TIP_SPEED,
+            'density_ratio',
+            Dynamic.Mission.INSTALLATION_LOSS_FACTOR,
+        ], rows=arange, cols=arange)
+        self.declare_partials('Thrust', [
+            Aircraft.Engine.PROPELLER_DIAMETER,
+        ])
+        self.declare_partials('propeller_efficiency', [
+            'adv_ratio',
+            'power_coefficient',
+            'thrust_coefficient',
+            'comp_tip_loss_factor',
+        ], rows=arange, cols=arange)
+        self.declare_partials('install_efficiency', [
+            'adv_ratio',
+            'power_coefficient',
+            'thrust_coefficient',
+            'comp_tip_loss_factor',
+            Dynamic.Mission.INSTALLATION_LOSS_FACTOR,
+        ], rows=arange, cols=arange)
+
+    def compute(self, inputs, outputs):
+        ctx = inputs['thrust_coefficient']*inputs['comp_tip_loss_factor']
+        outputs['thrust_coefficient_comp_loss'] = ctx
+        diam_prop = inputs[Aircraft.Engine.PROPELLER_DIAMETER][0]
```

Comment:
Why the 0 index on a scalar?

Reply:
Great question! More recent versions of numpy issue a warning like this if you do not use the `[0]` indexing:

```
/mnt/c/Users/John/Dropbox/git/om-aviary/aviary/subsystems/propulsion/hamilton_standard.py:507:
DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated,
and will error in future.
Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
```

Reply:
This made us realize we can change `math.sqrt` to `np.sqrt` to handle a similar deprecation warning, thank you

Reply:
Will this be an issue if we have different propellers on different engines?

Reply:
Yes; for now we'll only have one propeller type per plane